### PR TITLE
Add pause screen

### PR DESCRIPTION
**Pull request by charasyn**
This adds a pause screen. There is currently no help screen (only a placeholder) which should be fixed before the next demo release.

### CODE REVIEW

File: godotproject/Menus/InGameMenu/InGameMenu.gd [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	if screen_current == null:
+		# Hide ourselves
+		visible = false
+		# Unpause the other game logic
+		get_tree().paused = false
+	else:
+		# Set pause on SceneTree - this will pause the game logic
+		# (except for us, because we have our pause mode set to Process, not Inherit)
+		get_tree().paused = true
+		# Show ourselves
```
Comment (fheysel) [inline godotproject/Menus/InGameMenu/InGameMenu.gd]: by "show ourselves" you mean show the menu screen right?

File: godotproject/Menus/InGameMenu/PauseSubmenu.gd [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	0, 1, 2, 3,
+	2, 0, 1, 3
+]
+var debug_mode_input_index = 0
+
+func set_initial_focus():
+	first_focus.grab_focus()
+
+func activate_debug_mode():
+	print("Debug mode would be activated")
```
Comment (fheysel) [inline godotproject/Menus/InGameMenu/PauseSubmenu.gd]: This a TODO ?

File: godotproject/Menus/InGameMenu/PauseSubmenu.gd [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+func _process(_delta):
+	if Input.is_action_pressed("ui_cancel"):
+		var debug_mode_current_input = -1
```
Comment (fheysel) [inline godotproject/Menus/InGameMenu/PauseSubmenu.gd:28]: can we delete all this debug stuff or does it still need to be here?

Comment (charasyn) [reply godotproject/Menus/InGameMenu/PauseSubmenu.gd:28]: My plan for the "debug mode" was to basically have a set of inputs that we can put in on the pause menu that activates a debug mode where we can do administration tasks like properly shutting down the Pi. I did the code that would open up the debug menu but haven't actually made it do anything.

File: godotproject/Menus/InGameMenu/InGameMenu.gd (hunk 1 above)
Comment (charasyn) [reply godotproject/Menus/InGameMenu/InGameMenu.gd]: Yep, that's correct. How does this look?
```suggestion
		# Show ourselves (the in-game menu)
```

Comment (fheysel) [reply godotproject/Menus/InGameMenu/InGameMenu.gd]: yup! jsut wanted to be sure I understood what was happening there

File: godotproject/Menus/InGameMenu/PauseSubmenu.gd (hunk 3 above)
Comment (fheysel) [reply godotproject/Menus/InGameMenu/PauseSubmenu.gd:28]: ok cool could also be useful for enabling the players god_mode for testing purposes.

File: godotproject/Menus/InGameMenu/PauseSubmenu.gd (hunk 2 above)
Comment (charasyn) [reply godotproject/Menus/InGameMenu/PauseSubmenu.gd]: Yep! I think I will make a new version where this activates god mode on the player

**Comment by charasyn** [comment]
New functionality (featuring more hidden button presses) has been added allowing for the toggling of "god mode" on the player, as well as documentation to go along with it.

**Review by fheysel — APPROVED** [review]
Looks good! That's one one hell of a long cheat code! Could get a readme going with screen captures of all the menus?

**Pull request merged**